connector_salesforce/unit/importer_synchronizer.py

```diff
@@ -0,0 +1,389 @@
+        """ Call the convert function of the Mapper
+
+        in order to get converted record by using
+        mapper.data or mapper.data_for_create
+
+        :return: mapped dict of data to be used by
+                 :py:meth:``models.Model.create``
+        :rtype: dict
+        """
+        data = mapper.values(for_create=True, **kwargs)
+        self._validate_data(data)
+        return data
+
+    def _must_skip(self):
+        """Decide if the current record should
+        not be imported. By default it return
+        a False :py:class:`ImportSkipReason`
+
+        :return: a :py:class:`ImportSkipReason`
+        :rtype: :py:class:`ImportSkipReason`
+        """
+        return ImportSkipReason(False, None)
+
+    def _update(self, binding_id, data):
+        """Implement the update of a binding record
+        that is already existing in Odoo
+
+        :param binding_id: the id of the Odoo record to export
+        :type binding_id: int or long
+
+        :param data: mapped dict of data to be used by
+                     :py:meth:``models.Model.create``
+        """
+        self.session.write(self.model._name, binding_id, data)
+
+    def _validate_data(self, data):
+        """ Check if the values to import are correct
+
+        :param data: data to validate
+        Proactively check before the ``_create`` or
+        ``_update`` if some fields are missing or invalid.
+
+        Raise `InvalidDataError`
+        """
+        # we may want to return an NotImplementedError
+        return
+
+    def _import(self, binding_id):
+        """Try Import or update a binding record from Salesforce using REST API
+        :param binding_id: the current binding id in Odoo
+        :type binding_id: int or long or None
+        """
+        record_mapper = self.mapper.map_record(self.salesforce_record)
+        if binding_id:
+            # optimisation trick to avoid lookup binding
+            data = self._map_data_for_update(
+                record_mapper,
+                binding_id=binding_id,
+                backend_record=self.backend_record
+            )
+            self._update(binding_id, data)
+        else:
+            data = self._map_data_for_create(
+                record_mapper,
+                binding_id=binding_id,
+                backend_record=self.backend_record
+            )
+            binding_id = self._create(data)
+        self.binder.bind(self.salesforce_id, binding_id)
+        self._after_import(binding_id)
+
+    def run(self, salesforce_id, force_deactivate=False):
+        """Try to import or update a record on Salesforce using REST API
+        call required hooks and bind the record
+
+        :param salesforce_id: the current Salesforce UUID to import
+        :type binding_id: str
+
+        :param force_deactivate: If set to True it will force deactivate
+                           without calling _to_deactivate
+                           mostly use to save some REST calls
+        """
```

Comment:
That's the point, if the record has been deleted, there is nothing to import and no error should be raised.